### PR TITLE
feat: upgrade to USWDS v3.1.0

### LINKED ISSUE #2199: [chore] Upgrade USWDS to v3
**Which USWDS component(s) does this relate to?**
Dependencies

**Code line**
https://github.com/trussworks/react-uswds/blob/e36635f1ad3c7142f4860749e5392157d518e3d4/package.json#L52

**Further detail**
Hi! I wanted to check if we have plans to upgrade to [USWDS v3](https://designsystem.digital.gov/documentation/migration/) so that react-uswds will be v3 compatible? Apologies if there is already an issue for this elsewhere 🙏 and thanks in advance for any deets! Happy to try and take on the upgrade if helpful!

### PR DESCRIPTION
**Pull request by kleinschmidtj**
# Summary

This PR installs the new version of USWDS and follows the required steps in the [migration guide](https://designsystem.digital.gov/documentation/migration/).

- Gets rid of `uswdsResources.css` and moves the settings/overrides into `_uswds-theme.css` (as suggested by the guide.)
- Updates Sass entry point to use `@forward` syntax.
- Updates paths to USWDS assets

## Related Issues or PRs

Closes #2199 

## How To Test

Test by running `yarn build`, `yarn test`, and `yarn run storybook`. No errors should appear.

### CODE REVIEW

**Comment by jhancock532** [comment]
Thanks, I was in the middle of writing a issue for this, you've covered pretty much all the steps I'd considered in my review.

We may want to raise a separate issue & MR for [optimising the installation of USWDS in this project](https://designsystem.digital.gov/documentation/migration/#6-optimize-your-installation), or otherwise it would be nice to update the documentation to cover how users of the library can optimise their own projects. [I see you've raised an issue for documentation updates here](https://github.com/trussworks/react-uswds/issues/2202).

**Comment by haworku** [comment]
@jhancock532 I agree that we should have an issue for optimization updates and some place for discussion.  If you already have something you were thinking about go ahead  file an issue, it is very good to track this.

**Comment by haworku** [comment]
@kleinschmidtj thanks for the PR! 🌮 

Minor thing but I went ahead and changed the conventional commit prefix on your PR to `feat`. I think our users consider this a feature (we are moving up a major version in USWDS and consumers will all want to migrate alongside us if they haven't yet). Also `chore` commits don't show up in our changelog (you can see the settings for this in `package.json`, those can always be adjusted if the maintainer wants). Want to make sure this shows up. cc: @gidjin

File: package.json [hunk 1]
```diff
@@ -49,7 +49,7 @@
   "peerDependencies": {
     "react": "^16.x || ^17.x || ^18.x",
     "react-dom": "^16.x || ^17.x || ^18.x",
-    "uswds": "2.13.3"
+    "uswds": "3.1.0"
```
Comment (sawyerh) [inline package.json:52]: Should this be `@uswds/uswds`?

**Review by haworku — APPROVED** [review]
I just reviewed Happo ✅  ran through code change  and storybook ✅  , and  the testing steps ✅  so I approved based on that. 

I don't currently have access to a project I can pull this into to test it works but I feel based on how the storybook imports scss (which is similar to projects I have seen) it should be a good check there.

**Comment by aligg** [comment]
Thanks for working on this! @kleinschmidtj do you know if there are any plans to ship a release with these changes? 

(We're peckish for a react-uswds version that's uswds v3 compatible :) )

**Pull request merged**